src/d3a/models/strategy/external_strategies/storage.py

```diff
@@ -0,0 +1,261 @@
+import json
+import logging
+from d3a.models.strategy.external_strategies import IncomingRequest
+from d3a.models.strategy.storage import StorageStrategy
+from d3a.models.strategy.external_strategies import ExternalMixin, check_for_connected_and_reply
+
+
+class StorageExternalMixin(ExternalMixin):
+    """
+    Mixin for enabling an external api for the storage strategies.
+    Should always be inherited together with a superclass of StorageStrategy.
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.pending_requests = []
+
+    def event_activate(self):
+        super().event_activate()
```

Comment:
ExternalMixin class does not have a event_activate() method. Or am i missing something?

Reply:
This class is a mixin too though, and it depends on a normal strategy class. Therefore, this is not the event_activate of the EventMixin, but the event_activate of the strategy.